src/main/java/uk/gov/hmcts/divorce/caseworker/service/notification/GeneralEmailNotification.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.divorce.caseworker.service.notification;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.notification.EmailTemplateName;
+import uk.gov.hmcts.divorce.notification.NotificationService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.GeneralParties.APPLICANT;
+import static uk.gov.hmcts.divorce.divorcecase.model.GeneralParties.RESPONDENT;
+import static uk.gov.hmcts.divorce.divorcecase.model.LanguagePreference.ENGLISH;
+import static uk.gov.hmcts.divorce.notification.EmailTemplateName.GENERAL_EMAIL_OTHER_PARTY;
+import static uk.gov.hmcts.divorce.notification.EmailTemplateName.GENERAL_EMAIL_PETITIONER;
+import static uk.gov.hmcts.divorce.notification.EmailTemplateName.GENERAL_EMAIL_PETITIONER_SOLICITOR;
+import static uk.gov.hmcts.divorce.notification.EmailTemplateName.GENERAL_EMAIL_RESPONDENT;
+import static uk.gov.hmcts.divorce.notification.EmailTemplateName.GENERAL_EMAIL_RESPONDENT_SOLICITOR;
+import static uk.gov.hmcts.divorce.notification.NotificationConstants.APPLICANT_NAME;
+import static uk.gov.hmcts.divorce.notification.NotificationConstants.CCD_REFERENCE;
+import static uk.gov.hmcts.divorce.notification.NotificationConstants.GENERAL_EMAIL_DETAILS;
+import static uk.gov.hmcts.divorce.notification.NotificationConstants.GENERAL_OTHER_RECIPIENT_NAME;
+import static uk.gov.hmcts.divorce.notification.NotificationConstants.RESPONDENT_NAME;
+
+@Component
+@Slf4j
+public class GeneralEmailNotification {
+
+    @Autowired
+    private NotificationService notificationService;
+
+    public void send(final CaseData caseData, final Long caseId) {
+        log.info("Sending General Email Notification for case id: {}", caseId);
+
+        String emailTo = null;
+        EmailTemplateName templateId;
+
+        Map<String, String> templateVars = templateVars(caseData, caseId);
+
+        if (APPLICANT.equals(caseData.getGeneralEmail().getGeneralEmailParties())) {
+            if (caseData.getApplicant1().isRepresented()) {
+                log.info("Sending General Email Notification to petitioner solicitor for case id: {}", caseId);
+
+                emailTo = caseData.getApplicant1().getSolicitor().getEmail();
+                templateId = GENERAL_EMAIL_PETITIONER_SOLICITOR;
+                templateVars.put("solicitor name", caseData.getApplicant1().getSolicitor().getName());
+            } else {
+                log.info("Sending General Email Notification to petitioner for case id: {}", caseId);
+                emailTo = caseData.getApplicant1().getEmail();
+                templateId = GENERAL_EMAIL_PETITIONER;
+            }
+        } else if (RESPONDENT.equals(caseData.getGeneralEmail().getGeneralEmailParties())) {
+            if (caseData.getApplicant2().isRepresented()) {
```

Comment:
Could applicant2 ever be null and generalEmailParties be set to RESPONDENT?

Reply:
I think we will always have applicant 1 and applicant 2 so should not be an issue. When you create application I think we need both petitioner and solicitor details.
@linusnorton could you please confirm just to be on safer side :)

Reply:
Correct, we will always have an applicant2 after the application has been submitted and the caseworker can't see the case while it's in draft